common-custom-user-data-maven-extension/README.md

```diff
@@ -18,6 +18,15 @@ The additional tags, links and custom values captured by this extension include:
 See [CustomBuildScanEnhancements.java](./src/main/java/com/gradle/CustomBuildScanEnhancements.java) for details on what data is
 captured and under which conditions.
 
+#### Version compatibility
+
+This table details the version compatibility of the Common Custom User Data Maven extension with the Gradle Enterprise Maven extension.
+
+| Common Custom User Data Maven extension versions | Gradle Enterprise Maven extension versions |
+| ------------------------------------------------ | ------------------------------------------ |
+| `1.5.1`                                          | `1.10.1+`                                  |
+| `1.0` - `1.5.0`                                  | `1.0.0` - `1.10`                           |
```

Comment:
is it really the case that i cannot use CCUD extension 1.5.0 with GE extension 1.10.1?

Reply:
True. Fixed.

Reply:
@guylabs it's not clear to me why we need such a commpatibility table? why can I not use CCUD Maven extension 1.6 with GE Maven extension 1.9.2? (I understand that we would not get back anything for GradleEnterpriseApi), but that does not mean they are not compatible.

Reply:
//cc @facewindu

Reply:
Because of the import of the various API interfaces. Right on `CommonCustomUserDataMavenExtension`, it references them so in your example, it would not work at runtime, because not finding the import for the GradleEnterpriseApi interface.
For it to work, you'd need to return an `Object` from `ApiAccessor.lookupGradleEnterpriseApi`, and cast it back to its proper type, when not null, and when you really need it (in eg `GroovyScriptUserData.addToApis`)

Reply:
Right. I fogot that the GE Maven extension can be a compileOnly dependency, as in our build by default.

Reply:
I fixed the table. Resolving now.